hpx/util/register_locks.hpp

```diff
@@ -127,9 +127,6 @@ namespace hpx { namespace util
     inline void force_error_on_lock()
     {
     }
-    inline void enable_lock_detection()
-    {
-    }
     inline void ignore_lock(void const* /*lock*/)
```

Comment:
It's not needed in the case that lock detection is disabled at compile time. Why keep dead code?